include/core/LogMacros.h

```diff
@@ -73,25 +75,42 @@
 #ifdef LOG_WARN
 #undef LOG_WARN
 #endif
-#define LOG_WARN(message)                                                                   \
-    BOOST_LOG_STREAM_SEV(ml::core::CLogger::instance().logger(), ml::core::CLogger::E_Warn) \
-    LOG_LOCATION_INFO                                                                       \
-    message
+#define LOG_WARN(message)                                                       \
+    {                                                                           \
+        std::size_t countOfWarnMessages;                                        \
+        bool skipWarnMessage;                                                   \
+        std::tie(countOfWarnMessages, skipWarnMessage) =                        \
+            ml::core::CLogger::instance().throttler().skip(__FILE__, __LINE__); \
+        if (skipWarnMessage == false) {                                         \
+            BOOST_LOG_STREAM_SEV(ml::core::CLogger::instance().logger(),        \
+                                 ml::core::CLogger::E_Warn)                     \
+            LOG_LOCATION_INFO                                                   \
+            message << " [" << countOfWarnMessages << "]";                      \
```

Comment:
The format here is different to what's used for other repeated messages - see https://github.com/elastic/elasticsearch/blob/4de4c14b5b9e0ecb6be53dfe7bacf945e50efd55/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/logging/CppLogMessageHandler.java#L338

It will look inconsistent to people looking at the overall log file.

To make it consistent the format would be this:

```suggestion
            message << " | repeated [" << countOfWarnMessages << "]";                      \
```